### PR TITLE
Feature/resources e2e

### PR DESCRIPTION
**Pull request by curtwl**
# Description

Add e2e tests for resources controllers. 

## Issue link

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Tests

# How Has This Been Tested?

`yarn test:e2e resources`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: test/resources.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            },
+        },
+        select: {
+            userId: true,
+            voyageTeamId: true,
+        },
+    });
+};
+
+describe("ResourcesController (e2e)", () => {
```
Comment (cherylli) [inline test/resources.e2e-spec.ts:44]: could you please categories the tests by endpoints, you can look at auth, techs e2e tests for example

**Review by cherylli — CHANGES_REQUESTED** [review]
Could you structure the tests like 
describe (POST /voyages/:teamId/resources)
it(should return 201 if successful)
describe (should return 400 if request body is incomplete)
 it(missing title)
 it(missing url)
it (should return 401 if user is unauthorized)
etc

try to go through all responses we have on swagger

File: test/resources.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+        .expect(200);
+
+    return extractResCookieValueByKey(
+        res.headers["set-cookie"],
+        "access_token",
+    );
+};
+
+const findVoyageTeamId = async (email: string, prisma: PrismaService) => {
+    return await prisma.voyageTeamMember.findFirst({
```
Comment (cherylli) [inline test/resources.e2e-spec.ts:31]: await is redudant here, and line 45 (sorry my IDE is giving me a warning and I don't like seeing it 😂)

**Review by cherylli — CHANGES_REQUESTED** [review]
Looks good, but could you please add return code for all the `it` blocks? 
maybe like it `should return 201` ... or `should create a new resource and return 201`

I also suggest adding more `expect`s to check database records before and after operation
- for create, check that the new entry is in the database 
- for delete, check the number of records after deletion matches number of records before - number of records deleted
- for update, check that it's actually updated

Thanks!

**Comment by cherylli** [comment]
The tests are good, but it feels they are not consistent with other tests since we have all the 401 also included in specific endpoints in the other tests

**Review by JoshuaHinman — APPROVED** [review]
Tests passed.  Looks good!

**Comment by cherylli** [comment]
> > The tests are good, but it feels they are not consistent with other tests since we have all the 401 also included in specific endpoints in the other tests. I'm not sure if it's enough to request a change, what do you think?
> 
> I moved things around to make it simpler to have two beforeEachs with the different user logins, but now that I think about it, I could probably just call a helper function that logs out the first user and logs in the second for the 401 tests...

Thanks. You could look at one of these or add what you need https://github.com/chingu-x/chingu-dashboard-be/blob/dev/test/utils.ts

**Review by cherylli — APPROVED** [review]
this looks good to me, thanks!

**Review by timDeHof — APPROVED** [review]
Good job! All unit tests and e2e tests passed.

**Pull request merged**